forward_engineering/service/elastic_search_service/index.js

```diff
@@ -0,0 +1,145 @@
+const readline = require('readline');
+const fs = require('fs');
+const kibanaParser = require('../../script_parser/kibana_script_parser');
+const curlParser = require('../../script_parser/curl_script_parser');
+
+class ElasticSearchService {
+
+    _client = null;
+
+    /**
+     * @throws Error
+     * */
+    constructor(client) {
+        this._client = client;
+
+        this.applyToInstance = this.applyToInstance.bind(this);
+        this._executeScript = this._executeScript.bind(this);
+        this._insertExampleDocuments = this._insertExampleDocuments.bind(this);
+        this._insertExampleDocument = this._insertExampleDocument.bind(this);
+        this._insertExampleDocumentsFromFile = this._insertExampleDocumentsFromFile.bind(this);
+        this.testConnection = this.testConnection.bind(this);
+        this.close = this.close.bind(this);
+    }
+
+    /**
+     * @param script {string}
+     * @param entitiesData {{
+     *     [x: string]: {
+     *         name: string,
+     *         jsonData: string,
+     *         filePath?: string
+     *     }
+     * }}
+     * */
+    async applyToInstance(script, entitiesData) {
+        await this._executeScript(script);
+        const insertDocumentsPromises = Object.values(entitiesData)
+            .map((typeData) => {
+                const {filePath, jsonData } = typeData;
+                return this._insertExampleDocuments(JSON.parse(jsonData), filePath);
+            });
+        await Promise.all(insertDocumentsPromises);
+    }
+
+    async testConnection() {
+        await this._client.ping(undefined, { requestTimeout: 5000 });
+    }
+
+    /**
+     * @param script {string}
+     */
+    async _executeScript(script) {
+        let parsedData;
+        if (script.startsWith('curl')) {
+            parsedData = curlParser.parseCurlScript(script);
+        } else {
+            parsedData = kibanaParser.parseKibanaScript(script);
+        }
```

Comment:
Better to move parsing logic outside ElasticSearchService. This service should doesn't have business logic